commonutils/src/main/java/io/intrepid/commonutils/AndroidStringUtils.java

```diff
@@ -89,4 +98,37 @@ public static Spanned fromHtml(String source, int flags) {
             return Html.fromHtml(source);
         }
     }
+
+    /**
+     * Attempts to parse the given String to a Uri.
+     * This method exists because standard pattern matching functions (such as {@link Patterns#WEB_URL} treat addresses
+     * that omit a protocol (i.e. beginning with 'www') as valid.  However, if you try to actually use this Uri
+     * (e.g. calling CustomTabsIntent.launchUrl(Context, Uri)), it won't work.
+     * <p>
+     * This method will look for the case where the URL string is missing the protocol, and will prepend it if necessary.
+     *
+     * @param url - The String that you want to try parsing as a Uri
+     * @return A valid, well-formed Uri, or null if the url parameter was not valid
+     */
+    @Nullable
+    public static Uri parseUriFromString(@NonNull String url) {
+        URL validatedUrl = null;
+        try {
+            validatedUrl = new URL(url);
+        } catch (MalformedURLException e) {
+            try {
+                validatedUrl = new URL("http://" + url);
+            } catch (MalformedURLException ignored) {
```

Comment:
Maybe add warning/error logging in the catch statements and move the retry logic to a finally block? This is very stylistic and shouldn't affect code execution.